src/Game/CritterObject.cpp

```diff
@@ -482,8 +485,8 @@ void GameCritterObject::think()
             _orientation = hexagon()->orientationTo(movementQueue()->back());
             auto animation = _generateMovementAnimation();
             animation->setActionFrame(_running ? 2 : 4);
-            animation->addEventHandler("actionFrame",    std::bind(&GameCritterObject::onMovementAnimationEnded, this, std::placeholders::_1));
-            animation->addEventHandler("animationEnded", std::bind(&GameCritterObject::onMovementAnimationEnded, this, std::placeholders::_1));
+            EventManager::getInstance()->addHandler("actionFrame",    std::bind(&GameCritterObject::onMovementAnimationEnded, this, _1), animation);
+            EventManager::getInstance()->addHandler("animationEnded", std::bind(&GameCritterObject::onMovementAnimationEnded, this, _1), animation);
```

Comment:
This is procedural programming... it's BAD. Why don't stick with OOP approach (leaving addEventHandler method for objects)?